### PR TITLE
feat: reloader setup

### PR DESCRIPTION
**Pull request by Davidoutz**
## Description
Reloader is a controller that automatically watches for changes in ConfigMaps and Secrets and triggers rolling restarts  when required. This functionality ensures that applications deployed within a Kubernetes cluster always reflect the latest configuration without manual intervention.

## Motivation and Context
Applications often read configuration from ConfigMaps and Secrets at startup, and any changes to these configurations require manual intervention to take effect. This manual process involves either restarting the pods directly or reapplying the deployment, which adds unnecessary operational overhead and can lead to inconsistency if forgotten or mismanaged.

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: addons.tf [hunk 1]
```diff
@@ -202,3 +202,28 @@ module "cluster_secret_store" {
     module.addons
   ]
 }
+
+################################################################################
+# Reloader
+module "reloader" {
+  source = "./modules/addon"
+  create = var.create_addons && var.enable_reloader
```
Comment (swibrow) [inline addons.tf:210]: add a newline to make it a little more readable

File: addons.tf [hunk 2]
```diff
@@ -202,3 +202,28 @@ module "cluster_secret_store" {
...
+
+  chart         = "reloader"
+  chart_version = "1.0.121"
+  repository    = "https://stakater.github.io/stakater-charts"
+  description   = "Reloader"
+  namespace     = local.monitoring_namespace
```
Comment (swibrow) [inline addons.tf:216]: update namespace

File: files/helm/reloader/common.yaml [hunk 3]
```diff
@@ -0,0 +1,133 @@
+global:
```
Comment (swibrow) [inline files/helm/reloader/common.yaml:1]: delete this file. its not a good idea to add the default value

File: addons.tf [hunk 4]
```diff
@@ -202,3 +202,28 @@ module "cluster_secret_store" {
...
+
+  create_namespace = true
+
+  # https://github.com/stakater/Reloader/blob/master/deployments/kubernetes/chart/reloader/values.yaml
+  values = [
+    file("${path.module}/files/helm/reloader/common.yaml")
```
Comment (swibrow) [inline addons.tf:222]: I dont think you need any change anything so this can be removed.

**Review by swibrow — CHANGES_REQUESTED** [review]
small changes

File: files/helm/reloader/common.yaml (hunk 3 above)
Comment (Davidoutz) [reply files/helm/reloader/common.yaml:1]: Okey

File: addons.tf (hunk 1 above)
Comment (Davidoutz) [reply addons.tf:210]: Okey

File: addons.tf (hunk 4 above)
Comment (Davidoutz) [reply addons.tf:222]: Okay

**Review by swibrow — APPROVED** [review]
LGTM

**Pull request merged**